packages/babel-plugin-proposal-optional-chaining/src/index.js

```diff
@@ -24,8 +24,19 @@ export default declare((api, options) => {
     visitor: {
       "OptionalCallExpression|OptionalMemberExpression"(path) {
         const { scope } = path;
-        const parentPath = path.findParent(p => !p.isParenthesizedExpression());
+        // maybeParenthesized points to the outermost parenthesizedExpression
+        // or the path itself
+        let maybeParenthesized = path;
+        const parentPath = path.findParent(
+          p => !(p.isParenthesizedExpression() && (maybeParenthesized = p)),
+        );
```

Comment:
Nit: could you refactor this into something longer but easier to understand?

```suggestion
        const parentPath = path.findParent(p => {
          if (!p.isParenthesizedExpression()) return true;
          maybeParenthesized = p;
        });
```